hibernate-core/src/test/java/org/hibernate/test/hql/size/SizeTest.java

```diff
@@ -102,6 +113,31 @@ public void testSizeWithoutNestedPath() {
 		} );
 	}
 
+	@Test
+	public void testSizeWithoutNestedPathAddFetch() {
+		inTransaction( session -> {
+			List<Teacher> teachers = session.createQuery(
+					"select teacher from Teacher teacher left join fetch teacher.skills where size(teacher.skills) > 0",
+					Teacher.class
+			).getResultList();
+			assertEquals( countNumberOfJoins( sqlStatementInterceptor.getSqlQueries().get( 0 ) ), 2 );
+			assertEquals( 2, teachers.size() );
```

Comment:
The above assertion fails because the `Teacher` with 2 `Skill` objects in its collection ends up getting duplicated.

I believe this is a bug. I don't think it should be necessary to use `distinct` when adding a `join fetch`, but I am not 100% sure about that.

Reply:
I agree with you, this seems like a bug.

Reply:
@beikov, my guess for requiring ```distinct``` is that a ```join fetch``` is a kind of ```join```, and not something special with different rules. That's just my guess though.